File: September 2022/September_06.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public TreeNode pruneTree(TreeNode root) {
        return helper(root) ? root : null;
    }
    private boolean helper(TreeNode root){
        if(root == null)
            return false;
        boolean l = helper(root.left), r = helper(root.right);
        if(l == false)
            root.left = null;
        if(r == false)
            root.right = null;
        return (l || r || (root.val == 1));
    }
}

File: September 2022/September_06.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    bool helper(TreeNode* root){
        if(root == nullptr)
            return false;
        bool l = helper(root->left), r = helper(root->right);
        if(l == false)
            root->left = nullptr;
        if(r == false)
            root->right = nullptr;
        return (l || r || (root->val == 1));
    }
    
    TreeNode* pruneTree(TreeNode* root) {
        return helper(root) ? root : nullptr;
    }
};
